ci/release/apply_wheel_modifications.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+# Copyright (c) 2023, NVIDIA CORPORATION.
+#
+# Usage: bash apply_wheel_modifications.sh <new_version> <cuda_suffix>
+
+VERSION=${1}
+CUDA_SUFFIX=${2}
+
+# __init__.py versions
+sed -i "s/__version__ = .*/__version__ = \"${VERSION}\"/g" python/cudf/cudf/__init__.py
+sed -i "s/__version__ = .*/__version__ = \"${VERSION}\"/g" python/dask_cudf/dask_cudf/__init__.py
+sed -i "s/__version__ = .*/__version__ = \"${VERSION}\"/g" python/cudf_kafka/cudf_kafka/__init__.py
+sed -i "s/__version__ = .*/__version__ = \"${VERSION}\"/g" python/custreamz/custreamz/__init__.py
+sed -i "s/__version__ = .*/__version__ = \"${VERSION}\"/g" python/strings_udf/strings_udf/__init__.py
+
+# setup.py versions
+sed -i "s/version=.*,/version=\"${VERSION}\",/g" python/cudf/setup.py
+sed -i "s/version=.*,/version=\"${VERSION}\",/g" python/dask_cudf/setup.py
+sed -i "s/version=.*,/version=\"${VERSION}\",/g" python/cudf_kafka/setup.py
+sed -i "s/version=.*,/version=\"${VERSION}\",/g" python/custreamz/setup.py
+sed -i "s/version=.*,/version=\"${VERSION}\",/g" python/strings_udf/setup.py
+
+# cudf setup.py cuda suffixes
+sed -i "s/name=\"cudf\"/name=\"cudf${CUDA_SUFFIX}\"/g" python/cudf/setup.py
+sed -i "s/rmm/rmm${CUDA_SUFFIX}/g" python/cudf/setup.py
+sed -i "s/ptxcompiler/ptxcompiler${CUDA_SUFFIX}/g" python/cudf/setup.py
+sed -i "s/cubinlinker/cubinlinker${CUDA_SUFFIX}/g" python/cudf/setup.py
+
+# cudf pyproject.toml cuda suffixes
+sed -i "s/rmm/rmm${CUDA_SUFFIX}/g" python/cudf/pyproject.toml
+
+# dask_cudf setup.py cuda suffixes
+sed -i "s/name=\"dask-cudf\"/name=\"dask-cudf${CUDA_SUFFIX}\"/g" python/dask_cudf/setup.py
+# Need to provide the == to avoid modifying the URL
+sed -i "s/\"cudf==/\"cudf${CUDA_SUFFIX}==/g" python/dask_cudf/setup.py
```

Comment:
So for wheel releases, these rename packages to `xxx${CUDA_SUFFIX}` effectively replacing the `os.getenv` overriding of suffix with a sed-based solution.

As the comment on line 34 indicates, this is somewhat fragile since there might be other strings `setup.py` that we _don't_ want to modify that could match. Effectively we're trying to treat setup.py as a templated file and making a template substitution at a certain point.

I wonder if it would be less fragile, although feel free to push back, if the interpolation was done with sed with a halfway house from what we used to have and what we have now.

setup.py
```python

# overridden during wheel releases
cuda_suffix = ""
...
setup(..., requires=[f"rmm{cuda_suffix}", ...])
```

And then sed just runs:
```
sed -i "/cuda_suffix = \"\"/cuda_suffix = \"${CUDA_SUFFIX}\"/" setup.py
```

Which has a smaller chance of hitting false positives. WDYT?

Reply:
You are absolutely right. The reason I don't think that this change is worthwhile is that the imminent next step is to move away from setup.py entirely by migrating all of this information into a pyproject.toml file, a la https://github.com/rapidsai/rmm/pull/1151 (and I plan to replicate this change across RAPIDS). This change is part of the longer term plan to move us to more modern builders, but regarding the changes in this PR the effect will be that we will also have a more well-defined config (TOML) file to modify in the future rather than an arbitrarily formatted setup.py file, which should allow us to make more structured modifications in the future.

Reply:
Thanks, makes sense. In which case, good to go, although this lost the race with @brandon-b-miller so there are (hopefully trivial) conflicts to fix up :(.